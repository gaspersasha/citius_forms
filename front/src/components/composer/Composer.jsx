import React from 'react';
import PropTypes from 'prop-types';

const Composer = ({ components, children }) => (
  <>
    {components.reduceRight(
      (acc, Component) => (
        <Component>{acc}</Component>
      ),
      children
    )}
  </>
);

Composer.propTypes = {
  children: PropTypes.node.isRequired,
  components: PropTypes.arrayOf(PropTypes.elementType).isRequired,
};

export default Composer;
