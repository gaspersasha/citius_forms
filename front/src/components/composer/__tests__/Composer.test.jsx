import React from 'react';
import Composer from '../Composer';

describe('<Composer />', () => {
  const First = (props) => <div {...props} />;
  const Second = (props) => <div {...props} />;

  const props = {
    children: <div>children</div>,
    components: [First, Second],
  };

  it('renders without errors', () => {
    const wrapper = shallow(<Composer {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
