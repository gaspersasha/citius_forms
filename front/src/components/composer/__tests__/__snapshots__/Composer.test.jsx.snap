// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Composer /> renders without errors 1`] = `
<Fragment>
  <First>
    <Second>
      <div>
        children
      </div>
    </Second>
  </First>
</Fragment>
`;
